libosrmc/osrmc.cc

```diff
@@ -26,9 +26,30 @@ int osrmc_is_abi_compatible(void) { return osrmc_get_version() >> 16u == OSRMC_V
 /* API */
 
 struct osrmc_error final {
+  std::string code;
   std::string message;
 };
 
+static void osrmc_error_from_exception(const std::exception& e, osrmc_error_t* error) {
+  *error = new osrmc_error{"Exception", e.what()};
+}
+
+static void osrmc_error_from_json(osrm::json::Object* json, osrmc_error_t* error) try {
+  assert(json != nullptr);
```

Comment:
I just had a look at the call sites and it seems like we only use this internally in the C++ impl.

Then we could change this function to taking a `const osrm::json::Object&` (since we only need a read-only view for accessing the json values) and don't have to care about null pointers (since refs can't be null).

Reply:
Good point! Unfortunately it can't be `const` as, surprisingly`, indexing into `values` adds the key to the `unordered_map` if it doesn't exist. (This shouldn't actually matter in practise I don't think).